src/bugfender.ts

```diff
@@ -0,0 +1,364 @@
+        RnBugfender.setApiUrl(validatedOptions.apiURL);
+      }
+
+      if (typeof validatedOptions.baseURL !== 'undefined') {
+        RnBugfender.setBaseUrl(validatedOptions.baseURL);
+      }
+
+      if (validatedOptions.enableLogcatLogging) {
+        RnBugfender.enableLogcatLogging();
+      }
+
+      if (validatedOptions.logUIEvents) {
+        RnBugfender.enableUIEventLogging();
+      }
+
+      if (validatedOptions.registerErrorHandler) {
+        RnBugfender.enableCrashReporting();
+      }
+
+      if (validatedOptions.overrideConsoleMethods) {
+        this.overrideConsoleMethods.init(this.stringFormatter);
+      }
+
+      this.printToConsole.init(validatedOptions.printToConsole ?? true);
+
+      this.initialized = true;
+    }
+  }
+
+  /**
+   * Sets the name for the device. If the Device Name is not set, then the platform standard device name will be automatically sent
+   *
+   * @note This method has to be called prior to activate logger. Otherwise, an exception will be thrown
+   * @param {string} deviceName - Device name that will be sent to Bugfender.
+   */
+  public overrideDeviceName(deviceName: string) {
+    RnBugfender.overrideDeviceName(deviceName);
+  }
+
+  /**
+   * @returns Bugfender dashboard URL for the device
+   */
+  public getDeviceURL(): Promise<string> {
+    return RnBugfender.getDeviceUrl();
+  }
+
+  /**
+   * @returns Bugfender dashboard URL for the current session
+   */
+  public getSessionURL(): Promise<string> {
+    return RnBugfender.getSessionUrl();
+  }
+
+  /**
+   * Show a modal which asks for feedback. Once the user closes the modal or sends the feedback
+   * the returned promise resolves with the result.
+   *
+   * ```typescript
+   * Bugfender.getUserFeedback().then((result) => {
+   *     if (result.isSent) {
+   *         // User sent the feedback
+   *         // `result.feedbackURL` contains the Bugfender feedback URL
+   *     } else {
+   *         // User closed the modal without sending the feedback
+   *     }
+   * });
+   * ```
+   *
+   * @param options Options object to configure modal strings
+   * @returns Promise which resolves once the user closes the modal or sends the feedback
+   */
+  public async getUserFeedback(
+    options?: UserFeedbackOptions
+  ): Promise<UserFeedbackResult> {
+    return new Promise<UserFeedbackResult>(function (resolve) {
+      let defaultOptions = new DefaultUserFeedbackOptions();
+      if (typeof options === 'undefined') {
+        options = defaultOptions;
+      }
+      return RnBugfender.showUserFeedback(
+        options.title ?? defaultOptions.title,
+        options.hint ?? defaultOptions.hint,
+        options.subjectPlaceholder ?? defaultOptions.subjectPlaceholder,
+        options.feedbackPlaceholder ?? defaultOptions.feedbackPlaceholder,
+        options.submitLabel ?? defaultOptions.submitLabel,
+        options.closeLabel ?? defaultOptions.closeLabel
+      ).then(
```

Comment:
You could simplify this with the spread (`...`) operator. It's a handy way of merging two objects. It'll take the default values, and override them with those present in `options`. We default `options` to `{}` just in case it's `undefined`.

```suggestion
    options?: UserFeedbackOptions
  ): Promise<UserFeedbackResult> {
    return new Promise<UserFeedbackResult>((resolve) => {
      options = {
         ...new DefaultUserFeedbackOptions(),
         ...(options ?? {}),
      };

      return RnBugfender.showUserFeedback(
        options.title,
        options.hint,
        options.subjectPlaceholder,
        options.feedbackPlaceholder,
        options.submitLabel,
        options.closeLabel
      ).then(
```